www_src/pages/project/project.jsx

```diff
@@ -43,39 +43,74 @@ var Map = React.createClass({
       });
     });
   },
+
   componentDidMount: function () {
     var el = this.getDOMNode();
     var bounding = this.refs.bounding;
     var boundingEl = bounding.getDOMNode();
-    var startX, startY, deltaX, deltaY;
+    var scaleEl = this.refs.scaler.getDOMNode();
+    var isZoom, startX, startY, startDistance, deltaX, deltaY, currentZoom;
     var didMove = false;
+    var ZOOM_SENSITIVITY = 300;
 
     el.addEventListener('touchstart', (event) => {
       didMove = false;
       startX = event.touches[0].clientX;
       startY = event.touches[0].clientY;
       boundingEl.style.transition = 'none';
+      scaleEl.style.transition = 'none';
+
+      if (event.touches.length > 1) {
+        isZoom = true;
+        var dx = event.touches[1].clientX - startX;
+        var dy = event.touches[1].clientY - startY;
+        startDistance = Math.sqrt(dx*dx + dy*dy);
+      } else {
+        isZoom = false;
+      }
+
     });
 
     el.addEventListener('touchmove', (event) => {
       didMove = true;
+      if (isZoom && event.touches.length > 1) {
+        currentZoom = this.state.zoom;
+        var dx = event.touches[1].clientX - event.touches[0].clientX
+        var dy = event.touches[1].clientY - event.touches[0].clientY;
+        var distance = Math.sqrt(dx*dx + dy*dy);
 
-      var x = parseInt(this.state.camera.x);
-      var y = parseInt(this.state.camera.y);
+        currentZoom = currentZoom + ((distance - startDistance) / ZOOM_SENSITIVITY);
+        currentZoom = Math.min(Math.max(currentZoom, 0.25), 1);
 
-      deltaX = (event.touches[0].clientX - startX) / this.state.zoom;
-      deltaY = (event.touches[0].clientY - startY) / this.state.zoom;
+        scaleEl.style.transform = 'scale(' + currentZoom + ')';
+      } else if (!isZoom) {
+        var x = this.state.camera.x;
+        var y = this.state.camera.y;
 
-      var translation = 'translate(' + (x + deltaX) + 'px, ' + (y + deltaY) + 'px)';
-      boundingEl.style.transform = translation;
+        deltaX = (event.touches[0].clientX - startX) / this.state.zoom;
+        deltaY = (event.touches[0].clientY - startY) / this.state.zoom;
+
+        var translation = 'translate(' + (x + deltaX) + 'px, ' + (y + deltaY) + 'px)';
+        boundingEl.style.transform = translation;
+      }
 
     });
 
     el.addEventListener('touchend', (event) => {
       boundingEl.style.transition = '';
-      if (didMove) {
-        this.state.camera.x += deltaX;
-        this.state.camera.y += deltaY;
+      scaleEl.style.transition = '';
```

Comment:
same here. Reading from a DOM node = super useful. Changing things in a DOM node = rely on `setState` + `render`